projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6361,20 +6365,23 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      */
     endRowTransaction(commit: boolean, row: IgxRow) {
         row.newData = this.transactions.getAggregatedValue(row.id, true);
-
-        let args = row.createEditEventArgs();
+        const rowArgs = row.createEditEventArgs();
+        let rowEditArgs: IGridEditEventArgs;
 
         if (!commit) {
-            this.onRowEditCancel.emit(args);
+            this.rowEditExit.emit(rowArgs);
             this.transactions.endPending(false);
         } else {
-            args = this.gridAPI.update_row(row, row.newData);
+            rowEditArgs = this.gridAPI.update_row(row, row.newData);
+            this.rowEditExit.emit(rowArgs);
```

Comment:
The cancelation of rowEditExit has to be after the update of the row. So that's why it is below the update. rowEditExit should just prevent the editing to be closed but committed.